spec/javascripts/selectorSpec.js

```diff
@@ -37,4 +37,45 @@ describe("Suite for selector", function() {
     updateSelector("a", []);
     expect(d3.select("#aSelector").property("value")).toEqual("");
   });
+
+  it("should update clusterSelector when filterSelector==Cluster", function() {
+    d3.select("body")
+        .append("select")
+        .attr("id", "clusterSelector");
+    d3.select("body")
+        .append("select")
+        .attr("id", "filterSelector");
+
+    updateSelector("cluster", ["All", "cat", "dog"]);
+    updateSelector("filter", ["Cluster", "Zone"]);
+    d3.select("select#clusterSelector").property("value", "cat");
+
+    clusters = ["All", 0, 1];
+    updateFilter();
+    expect(d3.select("#clusterSelector").property("value")).toEqual("All");
```

Comment:
Could you add an inline comment explaining why you expect the value to be "All" here?